localizer.go

```diff
@@ -0,0 +1,59 @@
+package authboss
+
+import "context"
+
+type Localizer interface {
+	// Get the translation for the given text in the given context.
+	// If no translation is found, an empty string should be returned.
+	Localize(ctx context.Context, txt string, args ...any) string
+}
+
+// Translation constants
+const (
+	TxtSuccess = "success"
+
+	// Used in the auth module
+	TxtInvalidCredentials = "Invalid Credentials"
```

Comment:
In practice, I don't see a real difference between using the key or using the default string because anyone implementing would use the exported value anyway.

However, you are correct. The main reason for doing it this way is backwards compatibility.
We could use more traditional keys, but then we would have to explicitly maintain a default translator, and still fallback to the default when the provided translator does not have a translation.

I think that is extra complexity without any clear upside.